src/main/java/api/StudentApi.java

```diff
@@ -5,38 +5,71 @@
 
 import com.google.gson.Gson;
 import java.util.List;
+import java.util.stream.Collectors;
+
+import model.Lesson;
 import model.Student;
+import model.Subject;
+import model.Teacher;
 import persistence.StudentPersistence;
+import persistence.SubjectPersistence;
+import persistence.TeacherPersistence;
 import spark.Request;
 import spark.Response;
 
 public class StudentApi {
   private static final Gson gson = new Gson();
-  private static final StudentPersistence persistence = new StudentPersistence();
+  private final StudentPersistence studentPersistence;
+  private final TeacherPersistence teacherPersistence;
+  private final SubjectPersistence subjectPersistence;
+
+  public StudentApi(
+      StudentPersistence studentPersistence,
+      TeacherPersistence teacherPersistence,
+      SubjectPersistence subjectPersistence) {
+    this.studentPersistence = studentPersistence;
+    this.teacherPersistence = teacherPersistence;
+    this.subjectPersistence = subjectPersistence;
+  }
+
+  public List<api.dto.Student> getAll(Request request, Response response) {
+    return convertToDto(studentPersistence.getAll());
+  }
 
-  public List<Student> getAll(Request request, Response response) {
-    return persistence.getAll();
+  private List<api.dto.Student> convertToDto(List<Student> students) {
```

Comment:
Nie robiłbym tutaj takiego statycznego importowania
a) co w stytuacji kiedy będziemy mieli nasciane klasów student? Będziemy je importować tak, przecież wtedy nie bedziemy wiedzieć z czego dana klasa pochodzi. Poza tym klasy w danym pakiecie powinny mieć jedną unikalną nazwę.
b) czytelność to z punktu widzenia a bedzie bez potrzeby jak znowu bedziemy mieli naście klasów student.

Reply:
- to nie jest statyczne importowanie, tylko odnoszenie się nazwą pakietową
- nie będzie sytuacji z "naście klasów student" bo nie istnieje taki przypadek
- to nadal nie jest import tylko odnoszenie się nazwą pakietową i w tej nazwie widać "z czego" dana klasa pochodzi
- klasy w danym pakiecie powinny mieć unikalną nazwę i w tym przypadku mają ją
- nie ma wynikowej punktu a